File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/event/transfer/MoneyTransferReceivedEvent.java
package org.jmolecules.example.axonframework.domain.event.transfer;

import org.axonframework.serialization.Revision;

@Revision("1")
public record MoneyTransferReceivedEvent(
  String moneyTransferId,
  String targetAccountId,
  int amount
) {

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/bank/domain/moneytransfer/event/MoneyTransferReceivedEvent.kt
package org.jmolecules.example.axonframework.bank.domain.moneytransfer.event

import org.axonframework.serialization.Revision
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.AccountId
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.Amount
import org.jmolecules.example.axonframework.bank.domain.moneytransfer.type.MoneyTransferId

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyTransferReceivedEvent")
data class MoneyTransferReceivedEvent(
  val moneyTransferId: MoneyTransferId,
  val targetAccountId: AccountId,
  val amount: Amount
)
